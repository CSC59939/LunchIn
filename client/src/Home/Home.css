* {
  user-select: none !important;
}
.home-page {
  display: flex;
  flex-direction: column;
  max-width: 100% !important;
  margin-bottom: 5%;
}
.title-bar {
  padding-top: 5%;
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
  justify-content: center;
}
.home-page-content {
  display: flex;
  flex-direction: column;
  width: 50%;
  margin: 0 auto;
  margin-top: 3%;
  justify-content: center !important;
}
.login-form {
  flex: 1;
  margin: 0 20px;
  padding: 20px !important;
  border-radius: 25px !important;
}
.title-bar .title {
  font-size: 70px;
}
.title-bar .tag-line {
  font-size: 35px;
}
.push-down {
  margin-top: 10px !important;
}
@media only screen and (max-width: 1024px) {
  .title-bar .title {
    font-size: 50px;
  }
  .title-bar .tag-line {
    font-size: 30px;
  }
  .home-page-content {
    width: 65%;
  }
}
@media only screen and (max-width: 650px) {
  .title-bar .title {
    font-size: 40px;
  }
  .title-bar .tag-line {
    font-size: 20px;
  }
  .home-page-content {
    flex-direction: column;
    width: 90%;
  }
}
