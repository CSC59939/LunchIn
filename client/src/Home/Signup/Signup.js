import React, { Component } from "react";
import "./Signup.css";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  Divider,
  IconButton,
  InputAdornment,
  Snackbar,
  TextField,
  Typography,
  Tooltip
} from "@material-ui/core";
import CssBaseline from "@material-ui/core/CssBaseline";
import Visible from "@material-ui/icons/Visibility";
import Hidden from "@material-ui/icons/VisibilityOff";
import Check from "@material-ui/icons/CheckCircle";
import { firebase } from "../../Config";
import Geocode from "react-geocode";

const validation = {
  invalid: 0,
  valid: 1
};

export default class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userType: "user",
      name: "",
      email: "",
      password: "",
      address: "",
      lat: 0,
      lng: 0,
      showPassword: false,
      step1complete: false,
      step2complete: false,
      processing: false,
      notify: false,
      notifyMsg: "",
      message: "Show Password"
    };
    this.handleChange = this.handleChange.bind(this);
    this.createUser = this.createUser.bind(this);
    this.validate = this.validate.bind(this);
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  validate(type) {
    if (type === "createUser") {
      if (
        this.state.name === "" ||
        this.state.email === "" ||
        this.state.password === ""
      ) {
        return validation.invalid;
      }
    }
    return validation.valid;
  }
  createUser() {
    const isValid = this.validate("createUser");
    if (isValid === validation.valid) {
      this.setState({
        processing: true
      });

      Geocode.setApiKey(process.env.REACT_APP_GOOGLE_MAPS_API_KEY);
      console.log(this.state.address);
      Geocode.fromAddress(this.state.address).then(
        response => {
          const { lat, lng } = response.results[0].geometry.location;
          console.log(lat, lng);

          var setData = {
            type: this.state.userType,
            name: this.state.name,
            address: this.state.address,
            lat: lat,
            lng: lng
          };
          firebase
            .auth()
            .createUserWithEmailAndPassword(this.state.email, this.state.password)
            .then(() => {
              firebase
                .database()
                .ref(`Users/${firebase.auth().currentUser.uid}/`)
                .set(setData)
                .then(() => {
                  this.setState({
                    processing: false,
                    step1complete: true
                  }).then(() => {
                    this.props.history.push("/user/home");
                  });
                })
                .catch(error => {
                  this.setState({
                    notify: true,
                    notifyMsg: error.message,
                    processing: false,
                    step1complete: false
                  });
                });
            })
            .catch(error => {
              this.setState({
                notify: true,
                notifyMsg: error.message,
                processing: false,
                step1complete: false
              });
            });

        },
        error => {
          if (error.message === "Server returned status code ZERO_RESULTS") {
            console.error(error.message)
            this.setState({
              processing: false,
              notify: true,
              notifyMsg: "Invalid address. Please enter a valid address"
            });
          }
          if (error.message === "Server returned status code OVER_QUERY_LIMIT") {
            console.error(error.message)
            this.setState({
              processing: false,
              notify: true,
              notifyMsg: "Please try agian later"
            });
          }
        })

      // var setData = {
      //   type: this.state.userType,
      //   name: this.state.name,
      //   address: this.state.address
      // };
      // firebase
      //   .auth()
      //   .createUserWithEmailAndPassword(this.state.email, this.state.password)
      //   .then(() => {
      //     firebase
      //       .database()
      //       .ref(`Users/${firebase.auth().currentUser.uid}/`)
      //       .set(setData)
      //       .then(() => {
      //         this.setState({
      //           processing: false,
      //           step1complete: true
      //         }).then(() => {
      //           this.props.history.push("/user/home");
      //         });
      //       })
      //       .catch(error => {
      //         this.setState({
      //           notify: true,
      //           notifyMsg: error.message,
      //           processing: false,
      //           step1complete: false
      //         });
      //       });
      //   })
      //   .catch(error => {
      //     this.setState({
      //       notify: true,
      //       notifyMsg: error.message,
      //       processing: false,
      //       step1complete: false
      //     });
      //   });
    } else {
      this.setState({
        notify: true,
        notifyMsg: "Looks like you're missing stuff."
      });
    }
  }
  render() {
    return (
      <div className="signup-page">
        <CssBaseline />
        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={6000}
          message={this.state.notifyMsg}
        />
        <div className="signup-title-bar">
          <Typography variant="display2">LunchIn | Sign-Up</Typography>
          <Typography variant="subheading">
            Fill out the form below to sign-up for an account.
          </Typography>
        </div>
        <Divider />
        <div>
          <Card
            className="signup-page-content"
            style={{ marginTop: "5px" }}
            data-aos="slide-up"
          >
            <CardHeader
              title="Essentials"
              subheader="What we need to get started"
            />
            <CardContent>
              <TextField
                onChange={this.handleChange("name")}
                id="name"
                label="Full Name"
                required
                value={this.state.name}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                onChange={this.handleChange("email")}
                label="E-Mail"
                type="email"
                id="email"
                required
                value={this.state.email}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                onChange={this.handleChange("address")}
                id="address"
                label="Full Address"
                required
                value={this.state.address}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                disabled={this.state.step1complete || this.state.processing}
                id="password"
                label="Password"
                value={this.state.password}
                onChange={this.handleChange("password")}
                fullWidth
                required
                type={this.state.showPassword ? "text" : "password"}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <Tooltip title={this.state.message} placement="top">
                        <IconButton
                          onClick={() => {
                            if (this.state.message === "Show Password") {
                              this.setState({
                                showPassword: !this.state.showPassword,
                                message: "Hide Password"
                              });
                            } else if (this.state.message === "Hide Password") {
                              this.setState({
                                showPassword: !this.state.showPassword,
                                message: "Show Password"
                              });
                            }
                          }}
                        >
                          {this.state.showPassword ? <Visible /> : <Hidden />}
                        </IconButton>
                      </Tooltip>
                    </InputAdornment>
                  )
                }}
                className="push-down"
              />
              <Button
                onClick={() => {
                  this.createUser();
                }}
                fullWidth
                variant="raised"
                color="primary"
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              >
                <Check style={{ marginRight: "10px" }} />
                Sign Up
              </Button>
              <Button
                onClick={() => {
                  this.props.history.push(`/`);
                }}
                fullWidth
                variant="outlined"
                color="default"
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              >
                Already have an account? Sign in here!
              </Button>
            </CardContent>
          </Card>
          {this.state.step1complete ? (
            <Card className="push-down signup-page-content" data-aos="slide-up">
              <CardHeader
                title="Confirmation"
                subheader="Click done to create your account!"
              />
            </Card>
          ) : null}
        </div>
      </div>
    );
  }
}
