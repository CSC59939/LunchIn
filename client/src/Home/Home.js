import React, { Component } from "react";
import "./Home.css";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  IconButton,
  InputAdornment,
  LinearProgress,
  Snackbar,
  TextField,
  Typography,
  Tooltip
} from "@material-ui/core";
import CssBaseline from "@material-ui/core/CssBaseline";
import Visible from "@material-ui/icons/Visibility";
import Hidden from "@material-ui/icons/VisibilityOff";
import { firebase } from "../Config";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
      showPassword: false,
      notify: false,
      notifyMsg: "",
      processing: false,
      message: "Show Password"
    };
    this.handleChange = this.handleChange.bind(this);
    this.login = this.login.bind(this);
  }
  login() {
    this.setState({
      processing: true
    });
    firebase
      .auth()
      .signInWithEmailAndPassword(this.state.email, this.state.password)
      .then(
        resp => {
          this.props.history.push(`/user/home`);
          // wait for AuthStateChange
        },
        err => {
          this.setState({
            notify: true,
            notifyMsg: err.message,
            processing: false
          });
        }
      );
  }
  shouldComponentUpdate(nextProps, nextState) {
    if (nextProps.loggedIn !== this.state.processing) {
      nextState.processing = nextProps.loggedIn;
    }
    return true;
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };
  render() {
    return (
      <div className="home-page">
        <CssBaseline />
        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={3000}
          message={this.state.notifyMsg}
        />

        <div className="title-bar" data-aos="fade-up">
          <Typography
            component="h2"
            variant="h1"
            className="title"
            style={{ color: "#000000" }}
          >
            <strong>LunchIn</strong>
          </Typography>
          <Typography
            component="h2"
            variant="h3"
            className="tag-line"
            data-aos="fade-up"
          >
            Automatically Choose a Lunch Option For You
          </Typography>
        </div>
        <div className="home-page-content">
          <Card className="login-form" data-aos="slide-right">
            {this.state.processing ? (
              <LinearProgress className="loading" />
            ) : null}
            <CardHeader
              className="login-form-header"
              title="Sign In"
              subheader="Your Lunch Option Awaits"
            />
            <CardContent className="login-form-content">
              <TextField
                disabled={this.state.processing}
                className="email"
                id="email"
                label="E-mail"
                value={this.state.email}
                onChange={this.handleChange("email")}
                fullWidth
                required
                type="email"
              />
              <TextField
                disabled={this.state.processing}
                className="password"
                id="password"
                label="Password"
                value={this.state.password}
                onChange={this.handleChange("password")}
                fullWidth
                required
                type={this.state.showPassword ? "text" : "password"}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <Tooltip title={this.state.message} placement="top">
                        <IconButton
                          onClick={() => {
                            if (this.state.message === "Show Password") {
                              this.setState({
                                showPassword: !this.state.showPassword,
                                message: "Hide Password"
                              });
                            } else if (this.state.message === "Hide Password") {
                              this.setState({
                                showPassword: !this.state.showPassword,
                                message: "Show Password"
                              });
                            }
                          }}
                        >
                          {this.state.showPassword ? <Visible /> : <Hidden />}
                        </IconButton>
                      </Tooltip>
                    </InputAdornment>
                  )
                }}
              />
              <Button
                style={{ marginTop: 5, marginBottom: 5 }}
                disabled={this.state.processing}
                variant="contained"
                color="primary"
                fullWidth
                onClick={() => {
                  this.login();
                }}
                className="login"
              >
                Sign-In
              </Button>
              <Typography
                component="h2"
                style={{ marginTop: 5, marginBottom: 5 }}
                variant="subtitle1"
                className="signup-line"
              >
                Don't have an account? Signup now!
              </Typography>
              <Button
                style={{ marginTop: 5, marginBottom: 5 }}
                disabled={this.state.processing}
                variant="contained"
                color="secondary"
                fullWidth
                className="login"
                onClick={() => {
                  this.props.history.push("/sign-up");
                }}
              >
                Sign-Up
              </Button>
            </CardContent>
          </Card>
        </div>
      </div>
    );
  }
}
export default Home;
