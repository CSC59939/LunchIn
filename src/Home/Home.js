import React, { Component } from "react";
import "./Home.css";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  IconButton,
  InputAdornment,
  LinearProgress,
  Snackbar,
  TextField,
  Typography
} from "@material-ui/core";
import Visible from "@material-ui/icons/Visibility";
import Hidden from "@material-ui/icons/VisibilityOff";
import Next from "@material-ui/icons/CheckCircle";
import Signout from "@material-ui/icons/Cancel";
import { firebase } from "../Config";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      showPassword: false,
      notify: false,
      notifyMsg: "",
      processing: this.props.loggedIn
    };
    this.handleChange = this.handleChange.bind(this);
    this.login = this.login.bind(this);
  }
  login() {
    this.setState({
      processing: true
    });
    firebase
      .auth()
      .signInWithEmailAndPassword(this.state.username, this.state.password)
      .then(
        resp => {
          // wait for AuthStateChange
        },
        err => {
          this.setState({
            notify: true,
            notifyMsg: err.message,
            processing: false
          });
        }
      );
  }
  shouldComponentUpdate(nextProps, nextState) {
    if (nextProps.loggedIn !== this.state.processing) {
      nextState.processing = nextProps.loggedIn;
    }
    return true;
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };
  render() {
    return (
      <div className="home-page">
        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={3000}
          message={this.state.notifyMsg}
        />

        <div className="title-bar" data-aos="fade-up">
          <Typography variant="display4" className="title">
            <strong>LunchIn</strong>
          </Typography>
          <Typography
            variant="display2"
            className="tag-line"
            data-aos="fade-up"
          >
            Automatically Choose a Lunch Option For You
          </Typography>
        </div>
        <div className="home-page-content">
          <Card className="login-form" data-aos="slide-right">
            {this.state.processing ? (
              <LinearProgress className="loading" />
            ) : null}
            <CardHeader
              className="login-form-header"
              title="Sign In"
              subheader="Your Lunch Option Awaits"
            />
            {this.props.loggedIn === true ? (
              <CardContent className="login-form-content">
                <Typography className="push-down" variant="subheading">
                  Welcome, {firebase.auth().currentUser.displayName}!
                </Typography>
                <Button
                  fullWidth
                  color="primary"
                  variant="raised"
                  className="push-down"
                  onClick={() => {
                    this.props.history.push(`/user/home`);
                  }}
                >
                  <Next style={{ marginRight: "10px" }} />
                  Continue to Your Account
                </Button>
                <Button
                  fullWidth
                  color="secondary"
                  variant="raised"
                  className="push-down"
                  onClick={() => {
                    firebase.auth().signOut();
                  }}
                >
                  <Signout style={{ marginRight: "10px" }} />
                  Sign Out
                </Button>
              </CardContent>
            ) : (
              <CardContent className="login-form-content">
                <TextField
                  disabled={this.state.processing}
                  className="username"
                  id="username"
                  label="E-mail"
                  value={this.state.username}
                  onChange={this.handleChange("username")}
                  fullWidth
                  required
                  type="email"
                />
                <TextField
                  disabled={this.state.processing}
                  className="password"
                  id="password"
                  label="Password"
                  value={this.state.password}
                  onChange={this.handleChange("password")}
                  fullWidth
                  required
                  type={this.state.showPassword ? "text" : "password"}
                  InputProps={{
                    endAdornment: (
                      <InputAdornment position="end">
                        <IconButton
                          onClick={() => {
                            this.setState({
                              showPassword: !this.state.showPassword
                            });
                          }}
                        >
                          {this.state.showPassword ? <Visible /> : <Hidden />}
                        </IconButton>
                      </InputAdornment>
                    )
                  }}
                />
                <Button
                  style={{ marginTop: 5, marginBottom: 5 }}
                  disabled={this.state.processing}
                  variant="raised"
                  color="primary"
                  fullWidth
                  onClick={() => {
                    this.login();
                  }}
                  className="login"
                >
                  Sign-In
                </Button>
                <Typography
                  style={{ marginTop: 5, marginBottom: 5 }}
                  variant="subheading"
                  className="signup-line"
                >
                  Don't have an account? Signup now!
                </Typography>
                <Button
                  style={{ marginTop: 5, marginBottom: 5 }}
                  disabled={this.state.processing}
                  variant="raised"
                  color="secondary"
                  fullWidth
                  className="login"
                  onClick={() => {
                    this.props.history.push("/sign-up");
                  }}
                >
                  Sign-Up
                </Button>
              </CardContent>
            )}
          </Card>
        </div>
      </div>
    );
  }
}
export default Home;
