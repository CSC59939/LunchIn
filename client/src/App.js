import React, { Component } from "react";
import { Router, Route, Switch } from "react-router-dom";
import createHistory from "history/createBrowserHistory";
import UserRoutes from "./User/UserRoutes.js";
import Home from "./Home/Home.js";
import Signup from "./Home/Signup/Signup.js";
import { firebase } from "./Config";

class App extends Component {
  constructor(props) {
    super(props);
    // console.log(process.env);
    this.authListener = this.authListener.bind(this);
    this.firebaseListener = null;
    this.history = createHistory();
    this.state = {
      loggedIn: false,
      userType: null,
      lastLocation: "init"
    };
  }
  componentDidMount() {
    this.authListener();
    this.history.listen((loc, act) => {
      if (this.state.lastLocation === "/sign-up") {
        if (firebase.auth().currentUser) {
          firebase
            .database()
            .ref(`Users/${firebase.auth().currentUser.uid}/type`)
            .once("value", snap => {
              if (!snap.val()) {
                firebase.auth().currentUser.delete();
              }
            });
        }
        window.location.reload();
      }
      this.setState({
        lastLocation: loc.pathname
      });
    });
  }
  authListener() {
    this.fireBaseListener = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        this.setState({
          loggedIn: true
        });
        /*
                    User is logged in, check type and redirect here
                    How to redirect -- this.history.push('/sign-up');
                */
        firebase
          .database()
          .ref(`Users/${user.uid}/type`)
          .once("value", snap => {
            if (snap.val()) {
              const userType = snap.val();
              if (userType === "user") {
                this.history.push(`/user/home`);
              }
              this.setState({
                userType: userType
              });
            }
          });
      } else {
        this.setState({
          loggedIn: false,
          userType: null
        });
        this.history.push("/");
      }
    });
  }
  componentWillUnmount() {
    this.fireBaseListener && this.fireBaseListener();
    this.authListener = undefined;
  }
  render() {
    return (
      <Router history={this.history}>
        <div style={{ width: "100%", height: "100%" }}>
          <Switch>
            <Route
              exact
              path="/"
              render={props => {
                return (
                  <Home
                    {...props}
                    userType={this.state.userType}
                    loggedIn={this.state.loggedIn}
                  />
                );
              }}
            />
            <Route path="/sign-up" component={Signup} />
            <Route path="/user" component={UserRoutes} />>
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
