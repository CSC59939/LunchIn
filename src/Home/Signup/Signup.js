import React, { Component } from "react";
import "./Signup.css";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  Divider,
  IconButton,
  InputAdornment,
  Snackbar,
  TextField,
  Typography
} from "@material-ui/core";
import Visible from "@material-ui/icons/Visibility";
import Hidden from "@material-ui/icons/VisibilityOff";
import Next from "@material-ui/icons/PlayCircleFilled";
import Done from "@material-ui/icons/CheckCircle";
import { Link } from "react-router-dom";
import * as firebase from "firebase";

const validation = {
  invalid: 0,
  valid: 1,
  err_pwd_length: 2,
}


export default class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userType: "user",
      name: "",
      email: "",
      password: "",
      address: "",
      showPassword: false,
      step1complete: false,
      step2complete: false,
      processing: false,
      notify: false,
      notifyMsg: ""
    };
    this.handleChange = this.handleChange.bind(this);
    this.createUser = this.createUser.bind(this);
    this.finalizeUserUpdate = this.finalizeUserUpdate.bind(this);
    this.validate = this.validate.bind(this);
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };
  validate(type) {
    if (type === "createUser") {
      if (!this.state.name || !this.state.email || !this.state.password) {
        return validation.invalid;
      }
      if (this.state.password.length < 6) {
        return validation.err_pwd_length;
      }
    }
    return validation.valid;
  }
  createUser() {
    const isValid = this.validate("createUser");
    if (isValid === validation.valid) {
      this.setState({
        processing: true
      });
      firebase
        .auth()
        .createUserWithEmailAndPassword(this.state.email, this.state.password)
        .then(
          resp => {
            firebase.auth().currentUser.updateProfile({
              displayName: this.state.name
            });
          },
          this.setState({
            processing: false,
            step1complete: true
          }),
          err => {
            this.setState({
              notify: true,
              notifyMsg: err.message,
              processing: false
            });
          }
        );
    } else if(isValid === validation.err_pwd_length) {
      this.setState({
        notify: true,
        notifyMsg: "Password must at least 6 characters."
      });
    }
    else {
      this.setState({
        notify: true,
        notifyMsg: "Looks like you're missing stuff."
      });
    }
  }
  finalizeUserUpdate() {
    const isValid = this.validate("finalizeUser");
    if (isValid=== validation.valid) {
      this.setState({
        processing: true
      });
      var setData = {
        type: this.state.userType,
        name: this.state.name,
        address: this.state.address
      };
      firebase
        .database()
        .ref(`Users/${firebase.auth().currentUser.uid}/`)
        .set(setData)
        .then(
          () => {
            this.setState({
              processing: false,
              step2complete: true
            });
          },
          err => {
            this.setState({
              notify: true,
              notifyMsg: err.message,
              processing: false
            });
            return;
          }
        );
    } else if(isValid === validation.err_pwd_length) {
      this.setState({
        notify: true,
        notifyMsg: "Password must at least 6 characters."
      });
    }
     else {
      this.setState({
        notify: true,
        notifyMsg: "Looks like you're missing stuff.",
        processing: false
      });
    }
  }
  render() {
    return (
      <div className="signup-page">
        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={6000}
          message={this.state.notifyMsg}
        />
        <div className="signup-title-bar">
          <Typography variant="display2">LunchIn | Sign-Up</Typography>
          <Typography variant="subheading" style={{ marginTop: "10px" }}>
            Fill out the form below to sign-up for an account.
          </Typography>
        </div>
        <Divider />
        <div className="signup-page-content">
          <Card style={{ marginTop: "10px" }} data-aos="slide-up">
            <CardHeader
              title="Essentials"
              subheader="What we need to get started"
            />
            <CardContent>
              <TextField
                onChange={this.handleChange("name")}
                id="name"
                label="Full Name"
                required
                value={this.state.name}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                onChange={this.handleChange("address")}
                id="address"
                label="Full Address"
                required
                value={this.state.address}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                onChange={this.handleChange("email")}
                label="E-Mail"
                type="email"
                id="email"
                required
                value={this.state.email}
                fullWidth
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              />
              <TextField
                disabled={this.state.step1complete || this.state.processing}
                id="password"
                label="Password"
                value={this.state.password}
                onChange={this.handleChange("password")}
                fullWidth
                required
                type={this.state.showPassword ? "text" : "password"}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <IconButton
                        onClick={() => {
                          this.setState({
                            showPassword: !this.state.showPassword
                          });
                        }}
                      >
                        {this.state.showPassword ? <Visible /> : <Hidden />}
                      </IconButton>
                    </InputAdornment>
                  )
                }}
                className="push-down"
              />
              <Button
                onClick={() => {
                  this.createUser();
                }}
                fullWidth
                variant="raised"
                color="primary"
                className="push-down"
                disabled={this.state.step1complete || this.state.processing}
              >
                <Next style={{ marginRight: "10px" }} />
                Continue
              </Button>
            </CardContent>
          </Card>
          {this.state.step1complete ? (
            <Card className="push-down" data-aos="slide-up">
              <CardHeader
                title="Confirmation"
                subheader="Click done to create your account!"
              />
              <CardContent>
                <Button
                  onClick={() => {
                    this.finalizeUserUpdate();
                  }}
                  fullWidth
                  variant="raised"
                  color="primary"
                  className="push-down"
                  component={Link}
                  to="/"
                  disabled={this.state.step2complete || this.state.processing}
                >
                  <Done style={{ marginRight: "10px" }} />
                  Done
                </Button>
              </CardContent>
            </Card>
          ) : null}
        </div>
      </div>
    );
  }
}
