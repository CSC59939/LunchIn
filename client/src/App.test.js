import React from "react";
import { MemoryRouter } from "react-router-dom";
import App from "./App";
import { mount, shallow } from "enzyme";
import Home from "./Home/Home.js";
import Signup from "./Home/Signup/Signup.js";
import toJson, {
  shallowToJson,
  mountToJson,
  renderToJson
} from "enzyme-to-json";

describe("App Component", () => {
  it("should renders without crashing", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("converts shallow wrapper correctly", () => {
    const wrapper = shallow(<App />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("converts mount wrapper correctly", () => {
    const wrapper = shallow(<App />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
  it("converts render wrapper correctly", () => {
    const wrapper = shallow(<App />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });
});

describe("Router Test", () => {
  it("should redirect to Home (login) page successfully", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Home)).toHaveLength(1);
  });

  it("should redirect to Sign Up page successfully", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/sign-up"]}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Signup)).toHaveLength(0);
  });
});
