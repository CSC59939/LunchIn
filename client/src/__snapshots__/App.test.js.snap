// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component converts mount wrapper correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 3,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "uewq3o",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/sign-up"
      />
      <Route
        component={[Function]}
        path="/user"
      />
      &gt;
    </Switch>
  </div>
</Router>
`;

exports[`App Component converts render wrapper correctly 1`] = `null`;

exports[`App Component converts shallow wrapper correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 2,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "j2trb4",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/sign-up"
      />
      <Route
        component={[Function]}
        path="/user"
      />
      &gt;
    </Switch>
  </div>
</Router>
`;

exports[`App Component should renders without crashing 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/sign-up"
      />
      <Route
        component={[Function]}
        path="/user"
      />
      &gt;
    </Switch>
  </div>
</Router>
`;
